tests/test_api_calls.py

```diff
@@ -0,0 +1,25 @@
+import sys
+import os
+
+#append src dir to current dir. This has to be executed before importing the utils
+current_dir = os.path.dirname(os.path.abspath(__file__))
+sys.path.append(os.path.join(current_dir, '..', 'src'))
+
+import pytest
+from utils.ingestion_utils import OpenWeatherAPIGeocoding, FetchSecret
+
+
+
+#params
+@pytest.fixture(scope="session")    #fetch key once and store in fixture to avoid consuming calls when testing
+def api_key():
+    return FetchSecret("OpenWeatherAPIKeyYasin")
+
+
+def test_api_key(api_key):
+    print("API Key:",api_key)
+    assert api_key is not None  # make sure the key vault is returning something
+    assert isinstance(api_key, str)  #make sure you are receiving a string
+
+
```

Comment:
Great job on setting up these tests! Using pytest and the fixture for the API key is a smart way to avoid unnecessary calls. The path management is also well-handled.

Quick suggestion- 
Adding logging could help with troubleshooting.